Makefile.rule

```diff
@@ -30,18 +30,30 @@ BLASFEO_PATH = $(EXT_PATH)/blasfeo
 BLASFEO_VERSION = HIGH_PERFORMANCE
 # BLASFEO_VERSION = REFERENCE
 # BLASFEO_VERSION = BLAS_WRAPPER
+
 ## BLASFEO target
-BLASFEO_TARGET = X64_INTEL_HASWELL
+# BLASFEO_TARGET = X64_INTEL_HASWELL
 # BLASFEO_TARGET = X64_INTEL_SANDY_BRIDGE
 # BLASFEO_TARGET = X64_INTEL_CORE
-# BLASFEO_TARGET = GENERIC
+#
+# TARGET = X64_AMD_BULLDOZER
+# TARGET = X86_AMD_JAGUAR
+# TARGET = X86_AMD_BARCELONA
+#
+# TARGET = ARMV8A_ARM_CORTEX_A57
+# TARGET = ARMV8A_ARM_CORTEX_A53
+# TARGET = ARMV7A_ARM_CORTEX_A15
+# TARGET = ARMV7A_ARM_CORTEX_A7
+#
+BLASFEO_TARGET = GENERIC
 
 ## HPIPM path
 HPIPM_PATH = $(EXT_PATH)/hpipm
 #HPIPM_PATH = /home/gianluca/hpipm
+
 ## HPIPM target
-HPIPM_TARGET = AVX
-# HPIPM_TARGET = GENERIC
+# HPIPM_TARGET = AVX
```

Comment:
`AVX` should be the default

Reply:
I was copying the default from CMake,
just because this is the most portable option.
I thought was good to have same behavior Make/CMake,
but is not really important, neither to be portable.

For your information I add the option to specify your local override in `Makefile.local`
like in BLASFEO.

Just tell me what you prefer @giaf, @zanelli

Reply:
I would agree with @tmmsartor. I guess you could in principle automatically detect which instruction set is available on the machine where the code is built?

Reply:
How about something like this? https://github.com/magic-sph/magic/blob/master/cmake/FindSSE.cmake

Reply:
or this https://stackoverflow.com/questions/28939652/how-to-detect-sse-sse2-avx-avx2-avx-512-avx-128-fma-kcvi-availability-at-compile

Reply:
Nowadays 99.9% of PCs have AVX.

Reply:
And in general I prefer to have as default HASWELL + AVX because 1) it's faster, and otherwise people will use or benchmark against a slower version 2) more people using it means higher likelihood of detecting bugs (much less likely there are in GENERIC anyway) 3) if someone does not have HASWELL + AVX will switch himself or complain with us about something not working, which is easy to suggest to change

Reply:
About automatic detection, the correct way is to detect `cpuid` and get architecture information from that one, instead of detecting instruction set support. Some instruction sets may be supported on some architectures, but with bad performance.

If someone is volunteering to do that, OpenBLAS is doing it correctly. Otherwise in my opinion no detection as now is better.

Reply:
ok,  HASWELL+AVX is the default.
But with this setting there were some error on mac and windows,
yet to be investigated, for now on continuous integration I set again
the targets to GENERIC.